docs/source/how_to_add_a_component.rst

```diff
@@ -0,0 +1,112 @@
+How to Add a Component
+======================
+
+Intro
+-----
+
+To integrate a new component into the dashboard you need to add a folder (e.g.
+`maps`) in `utilities/dashboard/components`. The code that generates your component
+should be divided into three modules:
+
+- plotting module (e.g. `mapplot.py`), containing functions that process
+  a relatively "raw" dataset and returns the statistics to be displayed in the
+  plot, and functions that generate the component's plot.
+
+- `create_data.py`, containing functions that process the data (which may include
+  metadata, such as descriptive text) needed to create your component and returns
+  a dictionary of aggregate data.
+
+- `create_component.py`, containing functions that generate your component, which
+  may be a combination of plot and descriptive text, and possibly add callbacks
+  to make it interactive. Note that in the current dashboard layout
+  each component is displayed in a separate bokeh Tab.
+
+When you run the dashboard, `from_data_to_dashboard.py` will process the raw data
+and output the resulting dictionaries to a directory you specify.
+Note that the current version of the dashboard uses data from different raw datasets,
+which result in different dictionaries.
+`run_dashboard.py` will then use those dictionaries as inputs and assemble the
+components to create the dashboard.
+
+Therefore, to integrate your component in the existing code, you need to:
+
+  - **Update the data-processing step**, such that `from_data_to_dashboard.py`
+    additionally returns the dictionary you need to generate your component,
+  - **Update the dashboard-building step**, such that your component is displayed
+    together with the already existing Tabs.
+
+The two sections below contain more specific information on these two steps.
+The two overarching rules are: Make sure that your changes do not affect the
+already existing dashboard's components, and try to make your component as
+independent as possible.
+
+
+Data-processing
+---------------
+
+The result of this step will be a dictionary of aggregate data, to be passed
+to the functions that actually generate your component.
```

Comment:
"the functions that actually generate your component". Maybe you can be more specific here or give examples from the other components?

Reply:
I rewrote this section

Reply:
Very well done. Much more accessible for someone coming newly to the code! :+1: